### PR TITLE
Atomic-free JNI work

### PR DESCRIPTION
**Pull request by gacholio**
- re-enable JNI critical for atomic-free
- fix JNI check bug related to critical

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
jenkins test sanity zlinux jdk8

**Comment by charliegracie** [comment]
jenkins test sanity plinux jdk9

File: runtime/gc_glue_java/JNICriticalRegion.cpp [hunk 1]
```diff
@@ -36,6 +36,10 @@
 void
 MM_JNICriticalRegion::reacquireAccess(J9VMThread* vmThread, UDATA accessMask)
 {
+#if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
+	// Current thread must have entered the VM before acquiring exclusive
```
Comment (charliegracie) [inline runtime/gc_glue_java/JNICriticalRegion.cpp]: I am wondering if these asserts could be enabled all of the time? Not asking you to change them just thinking out loud.

File: runtime/gc_realtime/RealtimeAccessBarrier.cpp [hunk 2]
```diff
@@ -549,12 +545,12 @@ MM_RealtimeAccessBarrier::jniGetStringCritical(J9VMThread* vmThread, jstring str
 	} else {
 		// acquire access and return a direct pointer
 #if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
-		VM_VMAccess::inlineEnterVMFromJNI(vmThread);
-	  	vmThread->jniCriticalDirectCount += 1;
-		hasVMAccess = true;
-#else /* J9VM_INTERP_ATOMIC_FREE_JNI */
-		MM_JNICriticalRegion::enterCriticalRegion(vmThread);
+		if (!hasVMAccess) {
```
Comment (charliegracie) [inline runtime/gc_realtime/RealtimeAccessBarrier.cpp]: Also just making a note for myself to come look at the hasVMAccess.  Not sure how one thread entering here could have it and another thread not have it.

Comment (gacholio) [reply runtime/gc_realtime/RealtimeAccessBarrier.cpp]: I suspect this code can be refactored/optimized at some point.

File: runtime/gc_glue_java/JNICriticalRegion.cpp (hunk 1 above)
Comment (charliegracie) [reply runtime/gc_glue_java/JNICriticalRegion.cpp]: Sorry thought that was a VMAccess check lol.  My bad!

**Pull request merged**